distributed/comm/ucx.py

```diff
@@ -15,8 +15,13 @@
 from .utils import ensure_concrete_host, to_frames, from_frames
 from ..utils import ensure_ip, get_ip, get_ipv6, nbytes, log_errors
 
+from tornado.ioloop import IOLoop
 import ucp
 
+# TODO: remove dependency on Numpy and Numba
+import numpy as np
+import numba.cuda
```

Comment:
After going through this again, I'm curious about the use of Numpy to allocate buffers.  I was a little surprised by this, rather than using something like `memoryview`s (although presumably these are harder to create in pure Python).  Is this what you meant with this comment?

Reply:
Yes, right now the new UCX-py implementation doesn't support `memoryview`, just because it is hard to make right compared to `__array_interface__`. But I will add it soon :)